aspnetcore/blazor/lifecycle.md

```diff
@@ -0,0 +1,173 @@
+---
+title: ASP.NET Core Blazor lifecycle
+author: guardrex
+description: Learn how to use Razor component lifecycle methods in ASP.NET Core Blazor apps.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 11/23/2019
+no-loc: [Blazor]
+uid: blazor/lifecycle
+---
+# ASP.NET Core Blazor lifecycle
+
+By [Luke Latham](https://github.com/guardrex) and [Daniel Roth](https://github.com/danroth27)
+
+The Blazor framework includes synchronous and asynchronous lifecycle methods. Override lifecycle methods to perform additional operations on components during component initialization and rendering.
+
+### Component initialization methods
+
+<xref:Microsoft.AspNetCore.Components.ComponentBase.OnInitializedAsync*> and <xref:Microsoft.AspNetCore.Components.ComponentBase.OnInitialized*> execute code that initializes a component. These methods are only called one time when the component is first instantiated.
+
+To perform an asynchronous operation, use `OnInitializedAsync` and the `await` keyword on the operation:
+
+```csharp
+protected override async Task OnInitializedAsync()
+{
+    await ...
+}
+```
+
+> [!NOTE]
+> Asynchronous work during component initialization must occur during the `OnInitializedAsync` lifecycle event.
+
+For a synchronous operation, use `OnInitialized`:
+
+```csharp
+protected override void OnInitialized()
+{
+    ...
+}
+```
+
+### Before parameters are set
+
+<xref:Microsoft.AspNetCore.Components.ComponentBase.SetParametersAsync*> sets parameters supplied by the component's parent in the render tree:
+
+```csharp
+public override async Task SetParametersAsync(ParameterView parameters)
+{
+    await ...
+
+    await base.SetParametersAsync(parameters);
+}
+```
+
+<xref:Microsoft.AspNetCore.Components.ParameterView> contains the entire set of parameter values each time `SetParametersAsync` is called.
+
+The default implementation of `SetParametersAsync` sets the value of each property decorated with the `[Parameter]` or `[CascadingParameter]` attribute that has a corresponding value in the `ParameterView`. Parameters that don't have a corresponding value in `ParameterView` are left unchanged.
+
+If `base.SetParametersAync` isn't invoked, the custom code can interpret the incoming parameters value in any way required. For example, there's no requirement to assign the incoming parameters to the properties on the class.
+
+### After parameters are set
+
+<xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSetAsync*> and <xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSet*> are called when a component has received parameters from its parent and the values are assigned to properties. These methods are executed after component initialization and each time the parent component is rendered:
```

Comment:
The child component might not get rendered if the parameter values haven't changed.

```suggestion
<xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSetAsync*> and <xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSet*> are called when a component has received parameters from its parent and the values are assigned to properties. These methods are executed after component initialization and each time new parameter values are specified:
```